app/Models/Campaign.php

```diff
@@ -110,6 +110,17 @@ public function scopeWhereClosed($query)
             });
     }
 
+    /**
+     * Scope a query to only include campaigns containing specified cause within
+     * "cause" field.
+     */
+    public function scopeWhereHasCause($query, $cause)
+    {
+        // Regex matches on comma separated words to ensure precise match excepting first
+        // and last words (matching on only a following or preceding comma).
+        return $query->where('cause', 'regexp', '(^|,)'.$cause.'(,|$)');
```

Comment:
Is this possible to abuse? (Like, could someone enter a super-nasty regex as the "cause" and lock things up?)

Reply:
Oof, y'know I made a mental note to double-check regarding injection, and then completely. forgot to! I'll look into it!

Reply:
Added a line to sanitize the input (by stripping out all non-alphanumeric (aside from hyphens) characters) in https://github.com/DoSomething/rogue/pull/985/commits/75de1925df325650338bcccec570d42a3d66ea51